### ISSUE TITLE
Symphony Input Interface Error

### ISSUE THREAD

**Issue by cornejong** [deployphp/deployer#2299]
- Deployer version: ? unkown (latest, installed today)
- Deployment OS: Mac OS BigSur

After installing the Deployer i tried to run the init command (or any other for that matter) and i get the following error:
```
Fatal error: Declaration of Symfony\Component\Console\Input\Input::getArgument($name) must be compatible with Symfony\Component\Console\Input\InputInterface::getArgument(string $name) in phar:///usr/local/bin/dep/vendor/symfony/console/Input/Input.php on line 28
```

**Comment by antonmedv** [comment]
What php version?

**Comment by cornejong** [comment]
I'm using PHP 7.3.0.

**Comment by antonmedv** [comment]
What Deployer version?

**Comment by cornejong** [comment]
How can i find that out? I usually just run {command} -v for that, but for deployer i get the same error message. Is that easily found in the phar?

**Comment by antonmedv** [comment]
How did you install it?

**Comment by rparrett** [comment]
Experiencing the same issue after doing self-updating

```
18:27 $ php -v
PHP 7.1.23 (cli) (built: Feb 22 2019 22:19:32) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies

```
Issue persisted after reinstalling with

```
curl -LO https://deployer.org/deployer.phar
mv deployer.phar /usr/local/bin/dep
chmod +x /usr/local/bin/dep
```

**Comment by rparrett** [comment]
I've found that the latest version of deployer that doesn't produce this error is `6.7.0`.

**Comment by Tomsgu** [comment]
@rparrett you're using php 7.1 which is not compatible with deployer 6.8 see [composer.json 6.8.](https://github.com/deployphp/deployer/blob/v6.8.0/composer.json#L30). Seems like not documented BC break.

**Comment by cornejong** [comment]
```
➜  ~ curl -LO https://deployer.org/deployer.phar
mv deployer.phar /usr/local/bin/dep
chmod +x /usr/local/bin/dep

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 1037k  100 1037k    0     0  3089k      0 --:--:-- --:--:-- --:--:-- 3089k
➜  ~ dep

Fatal error: Declaration of Symfony\Component\Console\Input\Input::getArgument($name) must be compatible with Symfony\Component\Console\Input\InputInterface::getArgument(string $name) in phar:///usr/local/bin/dep/vendor/symfony/console/Input/Input.php on line 28
➜  ~ php -v
PHP 7.3.0 (cli) (built: Jan  2 2019 12:47:37) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.0-dev, Copyright (c) 1998-2018 Zend Technologies
➜  ~
```

**Comment by cornejong** [comment]
I tried to build the phar myself following [the instructions found here](https://deployer.org/docs/installation.html#own-builded-phar). But that didn't work for me.

Runs through all the files, but doesn't finish.
```php
// In bin/build at line 73. Never gets outputted to the terminal
echo "$pharName was created successfully.\n";
```
Also no ```deployer.phar``` file appears in the root.
```
➜  deployer git:(6.x) php bin/build
...
Add file: /src/Utility/Httpie.php
Add file: /src/Utility/ProcessOutputPrinter.php
Add file: /src/Utility/Reporter.php
Add file: /src/Utility/Rsync.php
➜  deployer git:(6.x) php deployer.phar
Could not open input file: deployer.phar
➜  deployer git:(6.x) ls
CHANGELOG.md       README.md          composer.json      recipe
CODE_OF_CONDUCT.md SPONSORS.md        composer.lock      src
KNOWN_BUGS.md      UPGRADE.md         docs               test
LICENSE            bin                phpunit.xml.dist   vendor
➜  deployer git:(6.x) cd bin
➜  bin git:(6.x) php dep
Deployer master
...
```

```$ bin/dep``` seems to work fine.

I also tried to build v7-master, but no luck.
Get basically the same result.

**Comment by cornejong** [comment]
Did the same for tag v6.8.0 as well now 😅
But still the same result:

```
➜  deployer git:(4e243a6) php bin/build
...
Add file: /src/Utility/Httpie.php
Add file: /src/Utility/ProcessOutputPrinter.php
Add file: /src/Utility/Reporter.php
Add file: /src/Utility/Rsync.php
➜  deployer git:(4e243a6) ls
CHANGELOG.md       README.md          composer.json      recipe
CODE_OF_CONDUCT.md SPONSORS.md        composer.lock      src
KNOWN_BUGS.md      UPGRADE.md         docs               test
LICENSE            bin                phpunit.xml.dist   vendor
➜  deployer git:(4e243a6) cd bin
➜  bin git:(4e243a6) php dep
Deployer master
...
```

**Comment by antonmedv** [comment]
Maybe this is related to BigSur? Can you test on other older mac?))

**Comment by cornejong** [comment]
Uhm, Yeah i still run catalina on my other computer, i'll give that a go today.

**Comment by cornejong** [comment]
I Just dove into ```bin/php``` (still on BigSur). And added the following to the top of the file:
```PHP
#!/usr/bin/env php
<?php

error_reporting(E_ALL);
ini_set('display_errors', true);
```
That gave me the following error:
```
Fatal error: Uncaught BadMethodCallException: unable to create temporary file in /Users/cornedejong/Development/Temp/deployer/deployer/bin/build:73
Stack trace:
#0 /Users/cornedejong/Development/Temp/deployer/deployer/bin/build(73): Phar->compressFiles(4096)
#1 {main}
  thrown in /Users/cornedejong/Development/Temp/deployer/deployer/bin/build on line 7
```

I Commented out the compressFiles method on line 73 in ```bin/build```
```PHP
// $phar->compressFiles(Phar::GZ);
```

Now the building of the phar works fine. Also ```deployer.phar``` runs without a hiccup. 
```
➜  deployer git:(4e243a6) php bin/build
...
Add file: /src/Utility/Reporter.php
Add file: /src/Utility/Rsync.php
deployer.phar was created successfully.
➜  deployer git:(4e243a6) php deployer.phar
Deployer dev-master
```

**Comment by cornejong** [comment]
I guess this is not a BigSur Issue, as @rparrett has the same issue but runs Mac OS 10.14.6 (Mojave) ([comment](https://github.com/deployphp/deployer/issues/2299#issuecomment-733404811)). But i'll give it a spin later today.

**Comment by antonmedv** [comment]
And build command has nozip option;)

**Comment by cornejong** [comment]
oh lol i didn't know that. but i fixed it in the pull request above. Maybe take a look at it.

**Comment by rparrett** [comment]
I did some testing with fresh `php:7.1-cli` and `php-7.2-cli` docker images and found that 7.1 produces the error first described in this ticket and 7.2 does not. Fair enough.

So I just downgraded to deployer 6.7.0 for now.

I do think it's unfortunate that a patch version (6.7.1) introduced an incompatibility with php, even if that version of php has reached its end of life. And also unfortunate that `dep self-update` let me update to a version that doesn't work with the version of PHP I'm using. And also that this incompatibility doesn't seem to be documented anywhere or mentioned in changelogs.

But it's way more unfortunate that I'm still running php 7.1, so it's not a big deal.

I'm not really following what's going on beyond that, but if there's something you want me to test with Mojave, let me know.

@cornejong Can we be 100% sure that the PHP the deployer phar is using is the same that is invoked when you did `php -v`? What's the output of `/usr/bin/env php -v`? Just an idea.
